src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java

```diff
@@ -134,6 +134,12 @@ public void updateColonyClient()
                 colonyId = citizen.getEntityData().get(DATA_COLONY_ID);
             }
 
+            if (colonyId == 0)
```

Comment:
Did you make sure these things don't happen atm regularily "just fine" ?

Reply:
ye tested a colony, this doesnt get hit normally

Reply:
Should probably add some logging here then when we do this